src/meshes/mesh_io.jl

```diff
@@ -225,11 +225,88 @@ function save_mesh_file(mesh::P4estMesh, output_directory, timestep, mpi_paralle
     return filename
 end
 
-# TODO: Implement this function as soon as there is support for this in `t8code`.
-function save_mesh_file(mesh::T8codeMesh, output_directory, timestep, mpi_parallel)
-    error("Mesh file output not supported yet for `T8codeMesh`.")
+# This routine works for both, serial and MPI parallel mode. The forest
+# information is collected on all ranks and then gathered by the root rank.
+# Since only the `levels` array of unsigned bytes is basicially independent per
+# rank it is not worth the effort to have a collective write to the HDF5 file.
+# Instead, `levels` gets gathered by the root rank and written in serial.
+function save_mesh_file(mesh::T8codeMesh, output_directory, timestep,
+                        mpi_parallel::Any)
```

Comment:
```suggestion
                        mpi_parallel)
```
Or maybe also `Union{False, True}`?

Reply:
Good point! I implemented Benedicts suggestion.